### ISSUE TITLE
smtp handler body html parse

### ISSUE THREAD

**Issue by ericxin1982** [Badgerati/Pode#1087]
### Question
A clear and concise description of your question.
Hi @Badgerati 

I have a new question, I try to parse the smtp body content, but when client sent, I cannot always ask them to change to plain text format, if they sent with html format, how I can parse them? Sorry for this stupid question.

Thanks
Eric Xin

**Comment by Badgerati** [comment]
Hi @ericxin1982,

Have you example of what the format the body is in, or how the email is being send? I would have expected the `$SmtpEvent.Email.Body` to contain the plain HTML from the client - but I can see this being different if the encoding/content-type isn't as expected 🤔

**Comment by ericxin1982** [comment]
Let me try again.
And I have another concern:
I sent my email from outlook 365 to this pode smtp server, outlook mail tracer showed the email has been delivered, but pode smtp server has no response, but some public email service sent, pode smtp server can receive, I am puzzled.

Thanks
Eric Xin

**Comment by Badgerati** [comment]
To be honest, I'm surprised Outlook can send email to a Pode SMTP server since it's not a public exchange! Where is the email being sent to, and how does it get to the Pode SMTP server?

If you have `New-PodeLoggingMethod -Terminal | Enable-PodeErrorLogging -Levels Error, Debug` set, do you see any errors on the terminal?

**Comment by ericxin1982** [comment]
I am running pode smtp server in the public internet,and give this IP to mx record for one speciifc domain, then the email can be delivered, ok, I would set up the log method and share you the output.

Thanks
Eric Xin

**Comment by SeriousMikeDE** [comment]
Apart from if it's possible to so this with pode I'd recommend rethinking this setup.

1. Running a webserver accessible from the internet directly sounds kinda scary
2. What I learned whenever I was trying to parse HTML is "never parse HTML"
Just my 2 cents but maybe there's a better solution for whatever you're trying to achieve

Greetings
Mike

**Comment by ericxin1982** [comment]
hi @Badgerati 

I got something from exchange sent.
![image](https://user-images.githubusercontent.com/9047436/230526940-4ce6dee6-39aa-4f96-aec4-fda58fa37756.png)

Can you see what that is.

Thanks
Eric Xin

**Comment by ericxin1982** [comment]
Hi @Badgerati 

I am new to github, so I do not know how to sumbit the change, but I have changed this bug.

This sentence below
headers.Add(match.Groups["name"].Value, match.Groups["value"].Value)

changed to the statement below.

if (headers.ContainsKey(match.Groups["name"].Value))
{
    headers[match.Groups["name"].Value] = headers[match.Groups["name"].Value].ToString + ";" + match.Groups["value"].Value;
}
else
{
    headers.Add(match.Groups["name"].Value, match.Groups["value"].Value);
}


And it works.
My code should not be perfect, but it works, I just share what I did.

Thanks
Eric Xin

**Comment by Badgerati** [comment]
Hi @ericxin1982,

Thanks for the code snippet 😄  I'll going to re-open this issue just so is stays on the radar and doesn't get lost for the next release 😉

**Comment by Badgerati** [comment]
Hi @ericxin1982,

I've applied a fix similar to yours above. The only difference is that when multiple headers are detected they'll be converted into a List object rather than appended as a string with `;` separators - as the Received header can also contain the `;` character.

**Comment by ericxin1982** [comment]
@Badgerati No problem, thanks for update. When 2.9 can be released? Thanks!

**Comment by Badgerati** [comment]
@ericxin1982 I'm hoping sometime this month.

**Comment by ericxin1982** [comment]
Thanks very much!
